backend/wmg/data/load_corpus.py

```diff
@@ -205,13 +215,20 @@ def save_raw(anndata_object: anndata.AnnData, group_name: str, global_var_index:
             )
             rankit_data = rankit_normalized_coo_sparse_raw_expression_matrix.data
 
-            array[rows, cols] = {"data": raw_data, "rankit": rankit_data}
+            # Filter out rankit values that were computed from expression values having raw count <= 3.
+
+            # TODO: Ideally, we would just *remove* these elements from rows,cols,rankit_data , but that would
+            #  require also adjusting the `obs` matrix and first_obs_idx, which are already updated. For now,
+            #  we will compute nnz without assuming all values are zero
+            rankit_data[rankit_data <= RANKIT_RAW_EXPR_COUNT_FILTERING_MIN_THRESHOLD] = 0
```

Comment:
Should use:
```
nonzero_mask = np.array(A[A.nonzero()] <= min_value)[0]
rows = A.nonzero()[0][nonzero_mask]
cols = A.nonzero()[1][nonzero_mask]
A[rows, cols] = 0
A.eliminate_zeros()
```
as it is drastically more performant.

Reply:
even if we're _not_ eliminating the zeros?

Reply:
I've updated the implementation quite a bit now. For one, it seems to filter out ~80% of the rankit values, and thus it is cheaper to _build a new_ sparse matrix from the values that we intend to _keep_ rather than zeroing-out and removing the elements we want to filter out. In any case, both strategies are implemented, with the caller able to decide on the strategy. Though currently, the code is unconditionally using the "build new" strategy, since _making_ the decision is too expensive!  

Performance-wise,  a 5GB test dataset consumes ~1min total wall time for this filtering operation, which seems reasonable.  @pablo-gar is verifying whether the extent of the filtering is correct, but it does match what the WMG prototype code filtered out.

Reply:
There's a slight discrepancy in numbers obtained from independent verification:

@atolopko-czi numbers from tabula sapiens endothelial:

filter <=3: 86% reduction
filter <=2: 77% reduction
filter <=1: 61% reduction

@pablo-gar  number from same data:
<=3 : 80.53559453292483 %
<=2 : 73.05934545284973 %
<=1 : 55.75715925770344 %

they are in the ballpark and this requires further investigation but it should not block this PR

Reply:
Re-verified,  and we're in agreement! 👏  I may have been using a diff dataset originally.  🤦   Validation notes in PR summary now.